wannierberri/__system.py

```diff
@@ -284,23 +289,130 @@ def set_symmetry(self,symmetry_gen=[]):
+                self.wannier_centers_reduced = self.wannier_centers_cart.dot(np.linalg.inv(self.real_lattice))
+#        self.wannier_centers_cart*=00.01
+#        self.wannier_centers_reduced*=00.01
+#        print ("Wannier_centers\n",self.wannier_centers_cart,self.wannier_centers_reduced)
+        if self.use_wcc_phase: 
+            if self.wannier_centers_cart is None:
+                raise ValueError("use_wcc_phase = True, but the wannier centers could not be detyermined")
+            if hasattr(self,'AA_R'):
+                AA_R_new = np.copy(self.AA_R)
+                AA_R_new[np.arange(self.num_wann),np.arange(self.num_wann),self.iR0,:] -= self.wannier_centers_cart
+            if hasattr(self,'BB_R'):
+                BB_R_new = self.BB_R.copy() - self.HH_R[:,:,:,None]*self.wannier_centers_cart[None,:,None,:]
+            if hasattr(self,'CC_R'):
+                norm = np.linalg.norm(self.CC_R - self.conj_XX_R(self.CC_R))
+                assert norm<1e-10 , f"norm={norm}"
+                assert hasattr(self,'BB_R') , "if you use CC_R, you need also BB_R"
+                T  =  self.wannier_centers_cart[:,None,None,:,None]*self.BB_R[:,:,:,None,:]
+                CC_R_new  =  self.CC_R.copy() + 1.j*sum(   
+                            s*( -T[:,:,:,a,b]   # -t_i^a * B_{ij}^b(R)
+                                -self.conj_XX_R(T[:,:,:,b,a])    # - B_{ji}^a(-R)^*  * t_j^b 
+                                +self.wannier_centers_cart[:,None,None,a]*self.HH_R[:,:,:,None] * self.wannier_centers_cart[None,:,None,b]  # + t_i^a*H_ij(R)t_j^b
+                            )
+                        for (s,a,b) in [(+1,alpha_A,beta_A) , (-1,beta_A,alpha_A)] )
+                norm = np.linalg.norm(CC_R_new - self.conj_XX_R(CC_R_new))
+                assert norm<1e-10 , f"norm={norm}"
+
+
+            # not sure if the following is correct (Stepan)
+            if hasattr(self,'SA_R'):
+                assert hasattr(self,'SS_R') , "if you use SA_R, you need also SS_R"
+                SA_R_new  =  self.SA_R.copy() - self.SS_R[:,:,:,:,None]*self.wannier_centers_cart[None,:,None,None,:]
+            if hasattr(self,'SHA_R'):
+                assert hasattr(self,'SH_R') , "if you use SA_R, you need also SH_R"
+                SHA_R_new  =  self.SHA_R.copy() - self.SS_R[:,:,:,:,None]*self.wannier_centers_cart[None,:,None,None,:]
+
+            for X in ['AA','BB','CC','SA','SHA']:
+                if hasattr(self,X+'_R'):
+                    vars(self)[X+'_R'] = locals()[X+'_R_new']
+
+            for X in ['SA','SHA','SR','SH','SHR']:
+                if hasattr(self,X+'_R'):
+                    pass
+#                    raise NotImplementedError(f"use_wcc_phases=True is not implemented for {X}_R")
+
+
+    @property
+    def iR0(self):
+        return self.iRvec.tolist().index([0,0,0])
+
+    @lazy_property.LazyProperty
+    def reverseR(self):
+        """maps the R vector -R"""
```

Comment:
R vector “to” -R? Or more precisely, it is the indices of R vectors that has -R in irvec, and the indices of the corresponding -R vectors.